<a href="https://colab.research.google.com/github/venkisamarth/python-programs/blob/main/Armstrong_number.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
number = int(input("Enter a number: "))
num_of_digits = len(str(number))
sum = 0
temp = number
while temp > 0:
    digit = temp % 10
    sum += digit * num_of_digits
    temp //= 10

if number == sum:
    print("The number is an Armstrong number.")
else:
    print("The number is not an Armstrong number.")


Enter a number: 153
The number is not an Armstrong number.


An Armstrong number is a number that is equal to the sum of its digits raised to the power of the number of digits. For example, 153 is an Armstrong number because 1^3 + 5^3 + 3^3 = 153.